File: scratch/copy_pages.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, '..', 'resources', 'js', 'pages');

const filesToCopy = [
    [path.join(baseDir, 'Admin', 'AdminAnnouncement.jsx'), path.join(baseDir, 'Admin', 'AdminFeaturedAlumni.jsx')],
    [path.join(baseDir, 'Admin', 'AdminAnnouncementCreate.jsx'), path.join(baseDir, 'Admin', 'AdminFeaturedAlumniCreate.jsx')],
    [path.join(baseDir, 'Admin', 'AdminAnnouncementEdit.jsx'), path.join(baseDir, 'Admin', 'AdminFeaturedAlumniEdit.jsx')],
    [path.join(baseDir, 'Admin', 'AdminAnnouncementView.jsx'), path.join(baseDir, 'Admin', 'AdminFeaturedAlumniView.jsx')],
    [path.join(baseDir, 'Coordinator', 'CoordinatorAnnouncement.jsx'), path.join(baseDir, 'Coordinator', 'CoordinatorFeaturedAlumni.jsx')],
    [path.join(baseDir, 'Coordinator', 'CoordinatorAnnouncementCreate.jsx'), path.join(baseDir, 'Coordinator', 'CoordinatorFeaturedAlumniCreate.jsx')],
    [path.join(baseDir, 'Coordinator', 'CoordinatorAnnouncementEdit.jsx'), path.join(baseDir, 'Coordinator', 'CoordinatorFeaturedAlumniEdit.jsx')],
    [path.join(baseDir, 'Coordinator', 'CoordinatorAnnouncementView.jsx'), path.join(baseDir, 'Coordinator', 'CoordinatorFeaturedAlumniView.jsx')],
];

filesToCopy.forEach(([src, dst]) => {
    let content = fs.readFileSync(src, 'utf-8');

    content = content.replace(/'admin\.announcement\.index'/g, "'admin.featured-alumni.index'");
    content = content.replace(/'admin\.announcement\.create'/g, "'admin.featured-alumni.create'");
    content = content.replace(/'admin\.announcement\.store'/g, "'admin.featured-alumni.store'");
    content = content.replace(/'admin\.announcement\.show'/g, "'admin.featured-alumni.show'");
    content = content.replace(/'admin\.announcement\.edit'/g, "'admin.featured-alumni.edit'");
    content = content.replace(/'admin\.announcement\.update'/g, "'admin.featured-alumni.update'");
    content = content.replace(/'admin\.announcement\.destroy'/g, "'admin.featured-alumni.destroy'");
    content = content.replace(/'admin\.announcement\.approve'/g, "'admin.featured-alumni.approve'");
    content = content.replace(/'admin\.announcement\.reject'/g, "'admin.featured-alumni.reject'");

    content = content.replace(/'coordinator\.announcement\.index'/g, "'coordinator.featured-alumni.index'");
    content = content.replace(/'coordinator\.announcement\.create'/g, "'coordinator.featured-alumni.create'");
    content = content.replace(/'coordinator\.announcement\.store'/g, "'coordinator.featured-alumni.store'");
    content = content.replace(/'coordinator\.announcement\.show'/g, "'coordinator.featured-alumni.show'");
    content = content.replace(/'coordinator\.announcement\.edit'/g, "'coordinator.featured-alumni.edit'");
    content = content.replace(/'coordinator\.announcement\.update'/g, "'coordinator.featured-alumni.update'");
    content = content.replace(/'coordinator\.announcement\.destroy'/g, "'coordinator.featured-alumni.destroy'");

    content = content.replace(/Announcements/g, "Featured Alumni");
    content = content.replace(/Announcement/g, "Featured Alumni");
    content = content.replace(/announcements/g, "featuredAlumni");
    content = content.replace(/announcement/g, "featuredAlumni");
    
    content = content.replace(/AdminFeatured Alumni/g, "AdminFeaturedAlumni");
    content = content.replace(/CoordinatorFeatured Alumni/g, "CoordinatorFeaturedAlumni");
    content = content.replace(/Featured AlumniCard/g, "FeaturedAlumniCard");
    content = content.replace(/Featured AlumniDeletePromptandConfirmation/g, "FeaturedAlumniDeletePromptandConfirmation");

    fs.writeFileSync(dst, content, 'utf-8');
});

console.log("Done!");

File: scratch/copy_pages.py
import os
import shutil

base_dir = r"d:\Github\tracer-study-system\resources\js\pages"

files_to_copy = [
    (rf"{base_dir}\Admin\AdminAnnouncement.jsx", rf"{base_dir}\Admin\AdminFeaturedAlumni.jsx"),
    (rf"{base_dir}\Admin\AdminAnnouncementCreate.jsx", rf"{base_dir}\Admin\AdminFeaturedAlumniCreate.jsx"),
    (rf"{base_dir}\Admin\AdminAnnouncementEdit.jsx", rf"{base_dir}\Admin\AdminFeaturedAlumniEdit.jsx"),
    (rf"{base_dir}\Admin\AdminAnnouncementView.jsx", rf"{base_dir}\Admin\AdminFeaturedAlumniView.jsx"),
    (rf"{base_dir}\Coordinator\CoordinatorAnnouncement.jsx", rf"{base_dir}\Coordinator\CoordinatorFeaturedAlumni.jsx"),
    (rf"{base_dir}\Coordinator\CoordinatorAnnouncementCreate.jsx", rf"{base_dir}\Coordinator\CoordinatorFeaturedAlumniCreate.jsx"),
    (rf"{base_dir}\Coordinator\CoordinatorAnnouncementEdit.jsx", rf"{base_dir}\Coordinator\CoordinatorFeaturedAlumniEdit.jsx"),
    (rf"{base_dir}\Coordinator\CoordinatorAnnouncementView.jsx", rf"{base_dir}\Coordinator\CoordinatorFeaturedAlumniView.jsx"),
]

for src, dst in files_to_copy:
    shutil.copy(src, dst)
    
    with open(dst, 'r', encoding='utf-8') as f:
        content = f.read()

    # Precise route string replacements
    content = content.replace("'admin.announcement.index'", "'admin.featured-alumni.index'")
    content = content.replace("'admin.announcement.create'", "'admin.featured-alumni.create'")
    content = content.replace("'admin.announcement.store'", "'admin.featured-alumni.store'")
    content = content.replace("'admin.announcement.show'", "'admin.featured-alumni.show'")
    content = content.replace("'admin.announcement.edit'", "'admin.featured-alumni.edit'")
    content = content.replace("'admin.announcement.update'", "'admin.featured-alumni.update'")
    content = content.replace("'admin.announcement.destroy'", "'admin.featured-alumni.destroy'")
    content = content.replace("'admin.announcement.approve'", "'admin.featured-alumni.approve'")
    content = content.replace("'admin.announcement.reject'", "'admin.featured-alumni.reject'")

    content = content.replace("'coordinator.announcement.index'", "'coordinator.featured-alumni.index'")
    content = content.replace("'coordinator.announcement.create'", "'coordinator.featured-alumni.create'")
    content = content.replace("'coordinator.announcement.store'", "'coordinator.featured-alumni.store'")
    content = content.replace("'coordinator.announcement.show'", "'coordinator.featured-alumni.show'")
    content = content.replace("'coordinator.announcement.edit'", "'coordinator.featured-alumni.edit'")
    content = content.replace("'coordinator.announcement.update'", "'coordinator.featured-alumni.update'")
    content = content.replace("'coordinator.announcement.destroy'", "'coordinator.featured-alumni.destroy'")

    # UI text replacements
    content = content.replace("Announcements", "Featured Alumni")
    content = content.replace("Announcement", "Featured Alumni")
    content = content.replace("announcements", "featuredAlumni")
    content = content.replace("announcement", "featuredAlumni")
    
    # Component name replacements (since we replaced Announcement with Featured Alumni, we need to fix the imports and component definitions)
    content = content.replace("AdminFeatured Alumni", "AdminFeaturedAlumni")
    content = content.replace("CoordinatorFeatured Alumni", "CoordinatorFeaturedAlumni")
    content = content.replace("Featured AlumniCard", "FeaturedAlumniCard")
    content = content.replace("Featured AlumniDeletePromptandConfirmation", "FeaturedAlumniDeletePromptandConfirmation")

    with open(dst, 'w', encoding='utf-8') as f:
        f.write(content)

print("Pages copied and modified.")
